spec.go

```diff
@@ -657,7 +658,18 @@ func (s *SpecValidator) validateParameters() *Result {
 			// TODO: should be done after param expansion
 			res.Merge(s.checkUniqueParams(path, method, op))
 
+			paramSchema, ok := s.schema.Definitions["parameter"]
+			if !ok {
+				panic("unexpected swagger schema: missing #/definitions/parameter")
+			}
+
 			for _, pr := range paramHelp.safeExpandedParamsFor(path, method, op.ID, res, s) {
+				// An expanded parameter must validate its schema (an unexpanded $ref always pass high-level schema validation)
+				schv := NewSchemaValidator(&paramSchema, s.schema, fmt.Sprintf("%s.%s.parameters.%s", path, method, pr.Name), s.KnownFormats, SwaggerSchema(true))
+				obj := swag.ToDynamicJSON(pr)
+				paramValidationResult := schv.Validate(obj)
+				res.Merge(paramValidationResult)
```

Comment:
I assume this will not lead to duplicate error entries for the same error detected twice.

Reply:
Good point. I should check that indeed. Let me give it a shot with another round of testing

Reply:
@youyuanwu added test to assert uniqueness. Done